src/webui/api/torrentscontroller.cpp

```diff
@@ -1041,3 +1041,70 @@ void TorrentsController::categoriesAction()
 
     setResult(categories);
 }
+
+void TorrentsController::addTagsAction()
```

Comment:
@Piccirello, we discussed error handlng in the early comments, please read them first. In a nutshell: since client-server interconnection is asynchronous, not all conflicts should cause an error, some can be safely ignored.

Reply:
I couldn’t find any substantive conversation about this above. The client processes API responses asynchronously, but that doesn’t mean our API should swallow errors. We should report the errors and allow the client to decide how to handle them. To me, that means 409 if all fail, 200 if all pass, and 200 with more info if some pass and some fail.